vfs/vfscache/item.go

```diff
@@ -581,7 +582,7 @@ func (item *Item) _store(ctx context.Context, storeFn StoreFn) (err error) {
 	// defer log.Trace(item.name, "item=%p", item)("err=%v", &err)
 
 	// Transfer the temp file to the remote
-	cacheObj, err := item.c.fcache.NewObject(ctx, item.name)
+	cacheObj, err := item.c.fcache.NewObject(ctx, item.c.fcache.(*local.Fs).LocalToStandardPath(toOSPath(item.name)))
```

Comment:
This is a fix for #7760  isn't it - it should say `Fixes #7760` in the commit messsage. Can you also write what the problem was and how it fixed it in the commit message too - thanks!

I think a better fix would be to create the local backend with the encoding that we want, something like this (untested)

```patch
diff --git a/vfs/vfscache/cache.go b/vfs/vfscache/cache.go
index 9bcfc90ee..0dc3b7630 100644
--- a/vfs/vfscache/cache.go
+++ b/vfs/vfscache/cache.go
@@ -212,7 +212,7 @@ func (c *Cache) createItemDir(name string) (string, error) {
 
 // getBackend gets a backend for a cache root dir
 func getBackend(ctx context.Context, parentPath string, name string, relativeDirPath string) (fs.Fs, error) {
-	path := fmt.Sprintf("%s/%s/%s", parentPath, name, relativeDirPath)
+	path := fmt.Sprintf(":local,encoding=%v:%s/%s/%s", encoder.OS, parentPath, name, relativeDirPath)
 	return fscache.Get(ctx, path)
 }
```

Reply:
Good idea!
Just need to place parameter in quotes.
Tested and working well.